sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessor.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.ServiceBus
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class ServiceBusSessionProcessor
+    {
+        private readonly ServiceBusProcessor _innerProcessor;
+
+        /// <summary>
+        /// The path of the Service Bus entity that the processor is connected to, specific to the
+        /// Service Bus namespace that contains it.
+        /// </summary>
+        public string EntityPath => _innerProcessor.EntityPath;
+
+        /// <summary>
+        /// Gets the ID to identify this client. This can be used to correlate logs and exceptions.
+        /// </summary>
+        /// <remarks>Every new client has a unique ID.</remarks>
+        internal string Identifier => _innerProcessor.Identifier;
+
+        /// <summary>
+        /// The <see cref="ReceiveMode"/> used to specify how messages are received. Defaults to PeekLock mode.
+        /// </summary>
+        public ReceiveMode ReceiveMode => _innerProcessor.ReceiveMode;
+
+        /// <summary>
+        /// The number of messages that will be eagerly requested from Queues or Subscriptions and queued locally without regard to
+        /// whether a processing is currently active, intended to help maximize throughput by allowing the receiver to receive
+        /// from a local cache rather than waiting on a service request.
+        /// </summary>
+        public int PrefetchCount => _innerProcessor.PrefetchCount;
+
+        /// <summary>
+        /// Indicates whether or not this <see cref="ServiceBusProcessor"/> has been closed.
+        /// </summary>
+        ///
+        /// <value>
+        /// <c>true</c> if the client is closed; otherwise, <c>false</c>.
+        /// </value>
+        ///
+        public bool IsProcessing => _innerProcessor.IsProcessing;
+
```

Comment:
`IsProcessing` would be true if the client is closed? It sounds weird. Did you mean something like `isClosed`/`isProcessorClosed`?

Reply:
Oh the docs need updating.